test/lib/tasks/bundlers/generateLibraryPreload.js

```diff
@@ -0,0 +1,131 @@
+const {test} = require("ava");
+const path = require("path");
+const chai = require("chai");
+chai.use(require("chai-fs"));
+const assert = chai.assert;
+
+
+const ui5Builder = require("../../../../");
+const builder = ui5Builder.builder;
+const libraryDPath = path.join(__dirname, "..", "..", "..", "fixtures", "library.d");
+const sapUiCorePath = path.join(__dirname, "..", "..", "..", "fixtures", "sap.ui.core");
+
+const recursive = require("recursive-readdir");
+
+const findFiles = (folder) => {
+	return new Promise((resolve, reject) => {
+		recursive(folder, (err, files) => {
+			if (err) {
+				reject(err);
+			} else {
+				resolve(files);
+			}
+		});
+	});
+};
+
+test("Build library.d with library preload", (t) => {
+	const destPath = "./test/tmp/build/library.d/preload";
+	const expectedPath = "./test/expected/build/library.d/preload";
+	const excludedTasks = ["*"];
+	const includedTasks = ["generateLibraryPreload"];
+
+	return builder.build({
```

Comment:
This call should be wrapped with a t.notThrows. `assert` is not integrated with ava, we use it because it throws when an assert fails. In that case, debugging issues is easier when telling ava that we the sad path expects this function to throw (i.e. the throw is not unexpected, but unwelcome).

Reply:
will use --> https://github.com/avajs/ava/blob/master/docs/03-assertions.md#notthrowsasyncnonthrower-message :)

Reply:
yes, just realized, will use notThrows. has to be adjusted when switching to a newser ava version.